src/state/reducers/messages.ts

```diff
@@ -88,7 +89,12 @@ export const hasUnseen: (
     getMessagesByBuddyId(buddyId)(appState),
     array.sort(ordMessage),
     array.last,
-    O.map(({ type, isSeen }) => type === 'Received' && isSeen === false),
+    O.map(
+      ({ type, isSeen }) =>
+        type === 'Received' &&
+        isSeen === false &&
+        !getIsBanned(buddyId)(appState),
+    ),
```

Comment:
i!sSeen would be easier to understand.

This is probably something that could be done on backend side already?

That map is way too complex. This should short-circuit befor if banned

Reply:
Like already before this whole pipe ? isBanned -> False

Reply:
There is actually a linter rule that disallows comparing booleans with boleans

Reply:
Added this to card